### PR TITLE
[Android][Accessibility] Prevent overwrite accessibilityLabel of BackButton 

### PR DESCRIPTION
**Pull request by retyui**
Issue: #7221, #7028

| Before | After |
|---|---|
| bad | good |

### CODE REVIEW

File: lib/android/app/src/main/java/com/reactnativenavigation/options/BackButton.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public boolean hasValue() {
 
     public void mergeWith(BackButton other) {
         if (!Constants.BACK_BUTTON_ID.equals(other.id)) id = other.id;
-        if (other.accessibilityLabel.hasValue()) accessibilityLabel = other.accessibilityLabel;
+        if (other.accessibilityLabel.hasValue() && !other.accessibilityLabel.equals(new Text("Navigate Up"))) accessibilityLabel = other.accessibilityLabel;
```
Comment (retyui) [inline lib/android/app/src/main/java/com/reactnativenavigation/options/BackButton.java:49]: Seem when user push a new screen `addToPushedChild` invoke `mergeOptions` with default options 


image

so, the custom value will be overwritten with a default value 
```java
this === child.options.topBar.buttons.back // My custom value ...
other === options.topBar.buttons.back  // Navigate Up

if (other.accessibilityLabel.hasValue()) accessibilityLabel = other.accessibilityLabel;
```

File: lib/android/app/src/main/java/com/reactnativenavigation/options/BackButton.java [hunk 2]
```diff
@@ -59,7 +59,7 @@ public void mergeWith(BackButton other) {
 
     void mergeWithDefault(final BackButton defaultOptions) {
         if (Constants.BACK_BUTTON_ID.equals(id)) id = defaultOptions.id;
-        if (!accessibilityLabel.hasValue()) accessibilityLabel = defaultOptions.accessibilityLabel;
+        if (!(accessibilityLabel.hasValue() && !accessibilityLabel.equals(new Text("Navigate Up")))) accessibilityLabel = defaultOptions.accessibilityLabel;
```
Comment (retyui) [inline lib/android/app/src/main/java/com/reactnativenavigation/options/BackButton.java:62]: Fix https://github.com/wix/react-native-navigation/issues/7028

image

**Review by swabbass — COMMENTED** [review]
Looks amazing!!. 

It would be cool to have basic unit tests.

**Comment by retyui** [comment]
@swabbass unit tests added

**Review by swabbass — APPROVED** [review]
AMAZING!!!!

**Comment by retyui** [comment]
that strange, e2e for iOS was failed

**Comment by swabbass** [comment]
> that strange, e2e for iOS was failed

triggered new build...it could be some flaky test. CC: @yogevbd

**Pull request merged**